social_django/utils.py

```diff
@@ -48,3 +50,15 @@ def wrapper(request, backend, *args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def maybe_require_post(func):
+    @wraps(func)
+    def wrapper(request, backend, *args, **kwargs):
+        require_post = getattr(settings, REQUIRE_POST, False)
+        if require_post and request.method != "POST":
+            return HttpResponseNotAllowed(["POST"])
+
+        return func(request, backend, *args, **kwargs)
+
+    return wrapper
```

Comment:
Maybe this could reuse require_POST from Django instead of reimplementing it?

Reply:
@nijel did you mean something like this?
```py
if require_post:
    return require_POST(func)(request, backend, *args, **kwargs)
```

Reply:
Yes, something like this.